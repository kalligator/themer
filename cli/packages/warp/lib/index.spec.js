const { render, renderInstructions } = require('./index');
const { colors } = require('../../colors-default');
const { load } = require('js-yaml');

let promisedFiles;

beforeAll(() => {
  promisedFiles = Promise.all(render(colors));
});

describe('render', () => {
  it('should render the expected number of files', async () => {
    const files = await promisedFiles;
    expect(files.length).toBe(2);
  });

  it('should render valid Warp theme yaml', async () => {
    const files = await promisedFiles;
    files.forEach((file) => {
      expect(load(file.contents.toString('utf8'))).toMatchSnapshot();
    });
  });
});

describe('renderInstructions', () => {
  it('should provide installation instructions', async () => {
    const files = await promisedFiles;
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
