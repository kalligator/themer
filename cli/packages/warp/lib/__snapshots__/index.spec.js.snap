// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render valid Warp theme yaml 1`] = `
Object {
  "accent": "#CC78FA",
  "background": "#282629",
  "details": "darker",
  "foreground": "#FFFCFF",
  "terminal_colors": Object {
    "bright": Object {
      "black": "#474247",
      "blue": "#66BFFF",
      "cyan": "#26C99E",
      "green": "#26C99E",
      "magenta": "#F553BF",
      "red": "#F28144",
      "white": "#FFFCFF",
      "yellow": "#FFD24A",
    },
    "normal": Object {
      "black": "#282629",
      "blue": "#66BFFF",
      "cyan": "#26C99E",
      "green": "#A4CC35",
      "magenta": "#F553BF",
      "red": "#FF4050",
      "white": "#E0DCE0",
      "yellow": "#FFD24A",
    },
  },
}
`;

exports[`render should render valid Warp theme yaml 2`] = `
Object {
  "accent": "#BF65F0",
  "background": "#FFFCFF",
  "details": "lighter",
  "foreground": "#282629",
  "terminal_colors": Object {
    "bright": Object {
      "black": "#474247",
      "blue": "#53A6E1",
      "cyan": "#1FC598",
      "green": "#1FC598",
      "magenta": "#EE4EB8",
      "red": "#F37735",
      "white": "#FFFCFF",
      "yellow": "#EEBA21",
    },
    "normal": Object {
      "black": "#282629",
      "blue": "#53A6E1",
      "cyan": "#1FC598",
      "green": "#97BD2D",
      "magenta": "#EE4EB8",
      "red": "#F03E4D",
      "white": "#E0DCE0",
      "yellow": "#EEBA21",
    },
  },
}
`;

exports[`renderInstructions should provide installation instructions 1`] = `
"
1. Copy your files (\`themer_dark.yaml\` and \`themer_light.yaml\`) to \`~/.warp/themes/\`
2. Launch Warp
3. Press \`command\`-\`P\` to open the Command Palette
4. Search and select \`Open Theme Picker\`
5. Search for \\"Themer Dark\\" or \\"Themer Light\\" and select it
6. Enjoy your new theme! âœ¨
"
`;
